projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1073,6 +1326,240 @@ export class IgxTimePickerComponent implements
         return this._ampmView.filter((ampm) => ampm !== '');
     }
 
+    /**
+     * @hidden
+     */
+    public hideOverlay(): void {
+        this.overlayService.hide(this._overlayId);
+    }
+
+    /**
+     * @hidden
+     */
+    public parseMask(preserveAmPm = true): string {
+        const prompts = this.promptChar + this.promptChar;
+        const amPm = preserveAmPm ? 'AM' : prompts;
+
+        return this.format.indexOf('tt') !== -1 ? `${prompts}:${prompts} ${amPm}` : `${prompts}:${prompts}`;
+    }
+
+    /**
+     * @hidden
+     */
+    public clear(): void {
+        if (this.collapsed) {
+            this.cleared = true;
+            this.isNotEmpty = false;
+
+            const oldVal = new Date(this.value);
+
+            this.displayValue = '';
+            this.value.setHours(0, 0);
+
+            if (oldVal.getTime() !== this.value.getTime()) {
+                const args: IgxTimePickerValueChangedEventArgs = {
+                    oldValue: oldVal,
+                    newValue: this.value
+                };
+                this.onValueChanged.emit(args);
+            }
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public onKeydown(event): void {
+        const alt = event.altKey;
```

Comment:
You don't need to create new variable here for the ALT key, because you use it only once. Also it will be better if you use the KEYS enum from core/utils.ts file for the 'switch' below.